flink-connector-mongodb/src/main/java/org/apache/flink/connector/mongodb/source/config/MongoReadOptions.java

```diff
@@ -0,0 +1,224 @@
+    public static class MongoReadOptionsBuilder {
+        private int fetchSize = SCAN_FETCH_SIZE.defaultValue();
+        private int cursorBatchSize = SCAN_CURSOR_BATCH_SIZE.defaultValue();
+        private boolean noCursorTimeout = SCAN_CURSOR_NO_TIMEOUT.defaultValue();
+        private PartitionStrategy partitionStrategy = SCAN_PARTITION_STRATEGY.defaultValue();
+        private MemorySize partitionSize = SCAN_PARTITION_SIZE.defaultValue();
+        private int samplesPerPartition = SCAN_PARTITION_SAMPLES.defaultValue();
+
+        /**
+         * Sets the number of documents should be fetched per round-trip when reading.
+         *
+         * @param fetchSize the number of documents should be fetched per round-trip when reading.
+         * @return this builder
+         */
+        public MongoReadOptionsBuilder setFetchSize(int fetchSize) {
+            checkArgument(fetchSize > 0, "The fetch size must be larger than 0.");
+            this.fetchSize = fetchSize;
+            return this;
+        }
+
+        /**
+         * Sets the batch size of MongoDB find cursor.
+         *
+         * @param cursorBatchSize the max batch size of find cursor.
+         * @return this builder
+         */
+        public MongoReadOptionsBuilder setCursorBatchSize(int cursorBatchSize) {
+            checkArgument(
+                    cursorBatchSize >= 0,
+                    "The cursor batch size must be larger than or equal to 0.");
+            this.cursorBatchSize = cursorBatchSize;
+            return this;
+        }
+
+        /**
+         * Set this option to true to prevent cursor timeout (10 minutes).
+         *
+         * @param noCursorTimeout Set this option to true to prevent cursor timeout (10 minutes)
+         * @return this builder
+         */
+        public MongoReadOptionsBuilder setNoCursorTimeout(boolean noCursorTimeout) {
+            this.noCursorTimeout = noCursorTimeout;
+            return this;
+        }
+
+        /**
+         * Sets the partition strategy.
+         *
+         * @param partitionStrategy the strategy of a partition.
+         * @return this builder
+         */
+        public MongoReadOptionsBuilder setPartitionStrategy(PartitionStrategy partitionStrategy) {
+            checkNotNull(partitionStrategy, "The partition strategy must not be null.");
+            this.partitionStrategy = partitionStrategy;
+            return this;
+        }
+
+        /**
+         * Sets the partition size of MongoDB split.
+         *
+         * @param partitionSize the memory size of a partition.
+         * @return this builder
+         */
+        public MongoReadOptionsBuilder setPartitionSize(MemorySize partitionSize) {
+            checkNotNull(partitionSize, "The partition size must not be null");
+            checkArgument(
+                    partitionSize.getMebiBytes() >= 1,
+                    "The partition size must be larger than or equals to 1mb.");
```

Comment:
why?

Reply:
Because the `splitVector` command's maxChunkSize minimum value was 1 mb. If we use parameters below 1mb, the command will fail. Also, chunks that are too small have no performance benefit for partitioning.

help for: splitVector Internal command.
examples:
  { splitVector : "blog.post" , keyPattern:{x:1} , min:{x:10} , max:{x:20}, maxChunkSize:200 }
  maxChunkSize unit in MBs
  May optionally specify 'maxSplitPoints' and 'maxChunkObjects' to avoid traversing the whole chunk